spark/spark-scala-parent/src/main/scala/org/apache/zeppelin/spark/BaseSparkScalaInterpreter.scala

```diff
@@ -79,9 +79,49 @@ abstract class BaseSparkScalaInterpreter(val conf: SparkConf,
     System.setProperty("scala.repl.name.line", ("$line" + this.hashCode).replace('-', '0'))
   }
 
-  protected def interpret(code: String, context: InterpreterContext): InterpreterResult
+  def interpret(code: String, context: InterpreterContext): InterpreterResult = {
+
+    def _interpret(code: String): scala.tools.nsc.interpreter.Results.Result = {
+      Console.withOut(interpreterOutput) {
+        interpreterOutput.setInterpreterOutput(context.out)
+        interpreterOutput.ignoreLeadingNewLinesFromScalaReporter()
+        context.out.clear()
+
+        val status = scalaInterpret(code) match {
+          case success@scala.tools.nsc.interpreter.IR.Success =>
+            success
+          case scala.tools.nsc.interpreter.IR.Error =>
+            if (new String(interpreterOutput.getInterpreterOutput.toByteArray)
+              .contains("value toDF is not a member of org.apache.spark.rdd.RDD")) {
+              // prepend "import sqlContext.implicits._" due to
+              // https://issues.scala-lang.org/browse/SI-6649
+              scalaInterpret("import sqlContext.implicits._\n" + code)
```

Comment:
Fixed

Reply:
IIRC, importing implicits has a bunch of [side effect to the code](http://spiridonov.pro/2015/10/14/scala-implicit-hell/)

also, we should try to avoid referencing sqlContext

why not only do this when needed, scala 2.10? maybe this will mask user code errors too.

Reply:
except when in newer spark it should use sparksession instead of sqlcontext, probably will cause some code to fail